banzai_floyds/tests/utils.py

```diff
@@ -0,0 +1,171 @@
+                    # add some random emission lines
+                    input_spectrum += strength * gauss(wavelengths.data[in_order],
+                                                       input_line, width) * profile
+                data[in_order] += input_spectrum
+
+        data[in_order] += background
+
+        if include_sky:
+            sky_wavelengths = np.arange(2500.0, 12000.0, 0.1)
+            sky_spectrum = np.zeros_like(sky_wavelengths) + sky_continuum
+            for line in SKYLINE_LIST:
+                line_spread = gauss(sky_wavelengths, line['wavelength'],
+                                    fwhm_to_sigma(line_widths[i]))
+                sky_spectrum += line['line_strength'] * line_spread * sky_normalization
+            # Make a slow illumination gradient to make sure things work even if the sky is not flat
+            illumination = 100 * gauss(slit_coordinates[in_order], 0.0, 48)
+            input_sky[in_order] = np.interp(wavelengths.data[in_order],
+                                            sky_wavelengths,
+                                            sky_spectrum) * illumination
+            data[in_order] += input_sky[in_order]
+    if fringe:
+        fringe_wave_number = 2.0 * np.pi / 30.0
+        expanded_orders = orders.new(expanded_order_height)
+        super_fringe_frame = 1.0 + 0.5 * (x2d / np.max(x2d)) * np.sin(fringe_wave_number * wavelengths.data)
+        super_fringe_frame[expanded_orders.data != 1] = 0.0
+        super_fringe_spline = fit_smooth_fringe_spline(super_fringe_frame, expanded_orders.data == 1)
+        in_red_order = orders.data == 1
+        input_fringe_frame = np.ones_like(data)
+
+        input_fringe_frame[in_red_order] = super_fringe_spline(np.array([x2d[in_red_order],
+                                                                        y2d[in_red_order] - fringe_offset]).T)
+
+        data *= input_fringe_frame
+
+    data = np.random.poisson(data.astype(int)).astype(float)
+    data += np.random.normal(0.0, read_noise, size=data.shape)
+    errors = np.sqrt(read_noise**2 + np.abs(data))
+
+    frame = FLOYDSObservationFrame([CCDData(data,
+                                            fits.Header({'DAY-OBS': '20230101',
+                                                         'DATE-OBS': '2023-01-01 12:41:56.11'}),
+                                            uncertainty=errors)],
+                                   'foo.fits')
+    frame.input_profile_centers = profile_centers
+    frame.input_profile_width = profile_width
+    frame.wavelengths = wavelengths
+    frame.orders = orders
+    frame.instrument = SimpleNamespace(site='ogg', camera='en02')
+    if include_sky:
+        frame.input_sky = input_sky
+    if fringe:
+        frame.fringe_wave_number = fringe_wave_number
+        frame.input_fringe_shift = input_fringe_shift
+        frame.input_fringe = input_fringe_frame
+    if include_super_fringe:
```

Comment:
Will break if `not fringe`. 
Might want to nest this with the above `if` or add an `and`.